### PR TITLE
Done

### PR DESCRIPTION
**Pull request by tejbans**

### CODE REVIEW

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+          
+
+    elsif kind =="0"
+            
+            Game.new(Player::Computer.new("X"), Player::Computer.new("O"), Board.new).play
+            puts "Do you want to play again (y/n)?."
+            play_again = gets.strip
+    end
+
+end
```
Comment (randallreedjr) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+require 'pry'
+class Board
+  attr_accessor:cells
+
+   def initialize
+    reset!
+   end
+
+   def reset!
+    @cells = [" "," "," "," "," "," "," "," "," "]
```
Comment (randallreedjr) [inline lib/board.rb]: Another way to instantiate an Array with `i` values is `Array.new(9, " ")` (see [Array.new](http://ruby-doc.org/core-2.2.0/Array.html#method-c-new)), where `9` is the number of elements in the array and `" "` is the default value for the elements. It's more concise and will prevent errors from repeatedly typing `" "`.

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+   end
+
+   def full?
+    self.cells.all?{|token| token == "X" || token == "O"}
+  end
+
+  def turn_count
+    self.cells.count{|t| t=="X" || t=="O"}
+  end
```
Comment (randallreedjr) [inline lib/board.rb]: Excellent choice of iterators for `full?` and `turn_count`

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+   def taken?(pos)
+    !(position(pos).nil? || position(pos) == " ")
+   end
+
+   def valid_move?(pos)
+    pos.to_i.between?(1,9) && taken?(pos) == false
```
Comment (randallreedjr) [inline lib/board.rb]: Avoid direct equality comparison with literal boolean values. You can reverse (or negate) the boolean with the `!` (bang) operator.

`pos.to_i.between?(1,9) && !taken?(pos)`

File: lib/players/computer.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      self.token == "X" ? "O" : "X"
+   end
+ 
+   def move(board)
+     return "5" if !board.taken?(5)
+     best_option(board)
+   end
+ 
+   def best_option(board)
+     win_move(board) || block_move(board) || strategic_move(board)
```
Comment (randallreedjr) [inline lib/players/computer.rb]: Nice method, very readable

**Comment by randallreedjr** [comment]
@tejbans Great job! I left you a few comments. Closing this PR for now, but feel free to re-open and respond if you would like to discuss any of them.

**Comment by tejbans** [comment]
Hello Randall,
	Thanks for the comments.
	I tried to refactor the code per comments and resubmit the project, but wasn’t able to?.
	Do I have to re-submit ?

thanks
Tejbans

**Comment by randallreedjr** [comment]
@tejbans You don't have to resubmit, the comments are just for your edification. However, if you would like to, you can try manually opening a new pull request from your github repo.

**Pull request closed without merging**